Feat/notification

Isaacvf-dev opened the pull request:
<img width="2924" height="1756" alt="image" src="https://github.com/user-attachments/assets/6a250cff-7a07-442c-b229-0a971ee6b494" />
<img width="2934" height="1752" alt="image" src="https://github.com/user-attachments/assets/24304722-427b-4b4a-8101-fe4c7f09a4df" />
<img width="1602" height="1474" alt="image" src="https://github.com/user-attachments/assets/2a4c97c9-c61e-4082-9c80-3a1b3594fb69" />
<img width="1068" height="1512" alt="image" src="https://github.com/user-attachments/assets/f2a93e02-a167-41b1-b750-4d832f2ba858" />

Isaacvf-dev committed:
Notification System

Added initial notification system layout

Isaacvf-dev committed:
Notification System

Connected the notification feature to the superbase's api

Isaacvf-dev committed:
Notification System

Refactoring and supabase connection

daniel-francelino commented on src/components/custom/SiteHeader.tsx:
```diff
@@ -1,22 +1,94 @@
+              <Button variant="ghost" size="sm" className="relative">
+                <Bell className="h-5 w-5" />
+                {unreadCount > 0 && (
+                  <span className="absolute -top-1 -right-1 bg-red-500 text-white text-xs font-bold rounded-full w-5 h-5 flex items-center justify-center">
```
Reduzir o tamanho para  w-4 h-4, ficou muito grande para o tamanho do icon.

<img width="150" height="94" alt="image" src="https://github.com/user-attachments/assets/0a95189c-c0b6-4f55-88a2-87c63ec46169" />

daniel-francelino commented on src/components/custom/SiteHeader.tsx:
```diff
@@ -1,22 +1,94 @@
+            </DropdownMenuTrigger>
+
+            <DropdownMenuContent align="end" className="w-80">
+              {loading && (
+                <div className="p-4 text-center text-muted-foreground text-sm">Carregando...</div>
+              )}
```
Utilizar o useTranslation e criar as traduções

daniel-francelino commented on src/components/custom/SiteHeader.tsx:
```diff
@@ -1,22 +1,94 @@
+                <div className="p-4 text-center text-red-500 text-sm">{error}</div>
+              )}
+
+              {!loading && notifications.length === 0 && (
+                <div className="p-4 text-muted-foreground text-sm text-center">
+                  Nenhuma notificação
+                </div>
+              )}
```
Utilizar o useTranslation e criar as traduções

daniel-francelino commented on src/components/custom/SiteHeader.tsx:
```diff
@@ -1,22 +1,94 @@
+                  onClick={markAllAsRead}
+                  className="text-muted-foreground cursor-pointer font-medium"
+                >
+                  Marcar todas como lidas
```
Utilizar o useTranslation e criar as traduções

daniel-francelino commented on src/components/custom/SiteHeader.tsx:
```diff
@@ -1,22 +1,94 @@
+                    <span className="font-medium flex-1">{notif.title}</span>
+                    {!notif.read_at && (
+                      <span className="text-xs bg-sidebar text-popover-foreground px-2 py-0.5 rounded-full">
+                        Novo
```
- Utilizar o useTranslation e criar as traduções;
- Adicionar opacity ao background 50% - 80%, alguma delas;
- Adicionar borda ao elemento;

daniel-francelino commented on src/components/custom/SiteHeader.tsx:
```diff
@@ -1,22 +1,94 @@
+                  className="flex flex-col items-start gap-1 p-3 cursor-pointer"
+                >
+                  <div className="flex items-center gap-2 w-full">
+                    <span className="font-medium flex-1">{notif.title}</span>
```
Adicionar semi bold no title

daniel-francelino commented on src/components/custom/SiteHeader.tsx:
```diff
@@ -1,22 +1,94 @@
+
+  const unreadCount = notifications.filter(n => !n.read_at).length;
+
+  const markAllAsRead = async () => {
```
- Dependendo da quantidade de notificação, vai fazer muitas requests para uma operação. Por exemplo se eu tiver 10 notificações não lidas, vai fazer 10 requests, enquanto deveria fazer somente 1 request;
- Deve também limitar a exibir somente 10 notificações, atualmente se eu tiver 99 notificações, vai ta exibindo as 99;

daniel-francelino commented on src/components/custom/SiteHeader.tsx:
```diff
@@ -1,22 +1,94 @@
-          <Button variant="ghost" size="sm" className="hidden sm:flex">
-            <Bell className="h-5 w-5" />
-          </Button>
+          <DropdownMenu>
```
- Adicionar "divider" entre as notificações, e mover o marcar todos para baixo, como o exemplo abaixo:

<img width="482" height="490" alt="image" src="https://github.com/user-attachments/assets/fbef55d5-4025-41cb-b457-03d9ffdc0e14" />

daniel-francelino commented on src/stores/notification.ts:
```diff
@@ -0,0 +1,41 @@
+  loading: false,
+  error: null,
+
+  fetchNotifications: async (userId: string) => {
+    set({ loading: true, error: null });
+    const { data, error } = await notificationApi.getNotifications(userId);
+    if (error) {
+      set({ error: error.message || "Erro ao carregar notificações", loading: false });
+    } else {
+      set({ notifications: data || [], loading: false });
+    }
+  },
+
+  markAsRead: async (notificationId: string) => {
+    try {
+      await notificationApi.markNotificationAsRead(notificationId);
+      set({
+        notifications: get().notifications.map((n) =>
+          n.id === notificationId ? { ...n, read_at: new Date().toISOString() } : n
+        ),
+      });
+    } catch (error) {
+      console.error("Erro ao marcar notificação como lida no store:", error);
+    }
+  },
```
Pode remover essas funções daqui do store, e deixar toda a lógica no src/lib/supabase/api/notification.ts;
Deixar somente o necessário para set notifications;

  setUser: (user) => set({ user }),

daniel-francelino reviewed (changes requested):
Atualiza a tua branch com o que tem na main, porque terá mais exemplos caso necessário olhar algum código;

Isaacvf-dev committed:
Merge branch 'main' into feat/notification

Isaacvf-dev committed:
Notification System

Fixed icon size and added translations

Isaacvf-dev committed:
Notification System

Refactoring logic to notification under API

Isaacvf-dev committed:
Notification System

Added separator between notifications, ui fixes

Isaacvf-dev committed:
Merge branch 'main' into feat/notification

Isaacvf-dev committed:
Notification System

Resolving conflicts

Isaacvf-dev committed:
Notification System

Showing only 5 notifications per turn. Fixing the logic for calling supabase api
